atoma-proxy/src/server/middleware.rs

```diff
@@ -687,16 +687,35 @@ pub async fn handle_locked_stack_middleware(
             // We need to acquire a new stack for the request, to be able to retry
             let user_id = request_metadata.user_id;
             let max_total_num_compute_units = request_metadata.max_total_num_compute_units;
-            // 1. Acquire a new stack for the request, this will also lock compute units for the new acquired stack
-            let selected_node_metadata = get_cheapest_node_and_acquire_new_stack(
+            // 1. Try to get a Stack from the state manager
+            let maybe_stack = get_node_metadata_from_state_manager(
```

Comment:
We should add some `debug` logs here 

```rust
tracing::debug!(
    "Attempting to acquire compute stack for user {} with model {} and max units {}",
    request_metadata.user_id,
    request_metadata.model_name,
    max_total_num_compute_units
);

// Later
tracing::debug!(
    "Successfully acquired stack {} on node {} for user {}",
    selected_node_metadata.stack_small_id,
    selected_node_metadata.selected_node_id,
    user_id
);
```

Reply:
This operation is not related to stack acquistion, it instead tries to read from the db.

Reply:
I added a few of these logs at the right place.